pysph/sph/equation.py

```diff
@@ -871,3 +871,15 @@ def __repr__(self):
         return '%s(groups=[[\n%s\n    ]]\n)' % (
             name, '], [\n'.join(kw)
         )
+
+    def __iter__(self):
```

Comment:
Don't you just need to do return `iter(self.groups)` here and not have to define a `__next__`

Reply:
removed it, now we check if the `equation` is a list or not in the logging method, thanks!